component/core.py

```diff
@@ -346,14 +346,18 @@ def component_by_name(self, name, model_name=None):
             work_context = self.work_on(model_name)
         return component_class(work_context)
 
-    def _lookup_components(self, usage=None, model_name=None):
+    def _lookup_components(self, usage=None, model_name=None, **kw):
         component_classes = self.components_registry.lookup(
             self.collection._name, usage=usage, model_name=model_name
         )
 
-        return [cls for cls in component_classes if cls._component_match(self)]
+        return [
+            cls
+            for cls in component_classes
+            if cls._component_match(self, usage=usage, model_name=model_name, **kw)
+        ]
```

Comment:
Should we degrade nicely in case an implementation have `_component_match` with the old signature?
That would force to have a less simple expression but would be more helpful for users:

```python
matching_components = []
for cls in component_classes:
    try:
        matching = cls._component_match(self, usage=usage, model_name=model_name, **kw)
    except TypeError as err:
        _logger.warn("The signature of %s._component_match must be changed to (self, usage=usage, model_name=model_name, **kw)", cls.__name__)
        matching =  cls._component_match(self)
    if matching:
        matching_components.append(cls)
return matching_components
```

Reply:
yeah, I though about it but them I thought: how many devs had ever customized this method? :sweat_smile: 

I'll apply this change anyway. Safer, less possible headaches.

Reply:
> how many devs had ever customized this method? sweat_smile

I did :laughing:

Reply:
fixed